client/components/settings/local_settings_modal_controller.js

```diff
@@ -0,0 +1,41 @@
+class LocalSettingsModalController {
+  constructor (localSettings) {
+    Object.assign(this, { localSettings })
+    this.settings = {}
+  }
+
+  $onInit () {
+    this.loadSettings()
+  }
+
+  loadSettings () {
+    this.settings = this.localSettings.getFromLocalStorage()
+    this.settingsData = {}
```

Comment:
`settingsData` doesn't represent what this value is. It should be `settingsByNamespace` or something of the sort.